pkg/kcp/nuke/reconciler.go

```diff
@@ -50,6 +58,11 @@ func (r *nukeReconciler) newAction() composed.Action {
 			deleteResources,
 			resourceStatusDeleting,
 			resourceStatusDeleted,
+			composed.BuildSwitchAction(
+				"NukeProviderSwitch",
+				nil,
+				composed.NewCase(focal.GcpProviderPredicate, gcpnuke.New(r.gcpStateFactory)),
+			),
```

Comment:
It's more readable with `composed.If`
```suggestion
			composed.If(
				focal.GcpProviderPredicate,
				gcpnuke.New(r.gcpStateFactory),
			),
```